### PR TITLE
feat: add tags to group

### PR DESCRIPTION
**Pull request by usmahm**
​
## Description

This pull request adds the ability to add tags to groups.

The affected endpoints include
GET - `/api/groups`  
GET - `/api/groups/:id`
GET - `/api/groups/user`
​

​
## Motivation and Context

Need ability to to add tags to groups
## How Has This Been Tested?

​It has been tested on Postman

​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: services/group.go [hunk 1]
```diff
@@ -113,43 +131,55 @@ func UpdateGroupById(id string, data *models.UpdateGroupRequest) (*models.Group,
...
-
+	err := query.
+		Preload("Tags").
+		Preload("Events", func(db *gorm.DB) *gorm.DB {
+			return db.Select("id")
+		}).
+		Preload("Members", "id").
+		Offset(offset).
+		Limit(limit).
+		Find(&groupDetailsList).Error
```
Comment (Badiru27) [inline services/group.go:153]: is event and group count no longer needed?

File: services/group.go [hunk 2]
```diff
@@ -113,43 +131,55 @@ func UpdateGroupById(id string, data *models.UpdateGroupRequest) (*models.Group,
...
+			"name":          group.Name,
+			"image":         group.Image,
+			"created_at":    group.CreatedAt,
+			"updated_at":    group.UpdatedAt,
+			"tags":          group.Tags,
+			"members_count": len(group.Members),
+			"events_count":  len(group.Events),
+		}
+		response = append(response, groupMap)
+	}
```
Comment (Badiru27) [inline services/group.go:171]: What happens when group hits 1 million records?  This should be done from the DB layer

**Review by Badiru27 — CHANGES_REQUESTED** [review]

File: services/group.go (hunk 2 above)
Comment (usmahm) [reply services/group.go:171]: I figured since we're paginating, it shouldn't be a problem looping 10 times, even though it's not the best approach

Comment (Badiru27) [reply services/group.go:171]: True. 10 items won't be much of a big problem

**Review by Badiru27 — APPROVED** [review]

**Pull request merged**